File: aufgabe03/manschwa/IRC.rb
require "socket"

server = "irc.freenode.net"
port = "6667"
nick = "ManoBot"
channel = "#codingchaos"

s = TCPSocket.open(server, port)
print("addr: ", s.addr.join(":"), "\n")
print("peer: ", s.peeraddr.join(":"), "\n")
s.puts "USER testing 0 * Testing"
s.puts "NICK #{nick}"
s.puts "JOIN #{channel}"
#s.puts "PRIVMSG #{channel} :dumdidum"

until s.eof? do
  msg = s.gets
  realmsg = msg.split(':')[2]
  puts realmsg
  if realmsg === "!say"
    s.puts "PRIVMSG #{channel} :hello!"
  end

  puts msg
end

File: aufgabe03/jguese/irc.py
import socket

network = 'irc.freenode.net'
port = 6667
irc = socket.socket ( socket.AF_INET, socket.SOCK_STREAM )
irc.connect ( ( network, port ) )
print irc.recv ( 4096 )


irc.send ( 'USER jguese Hostname Servername Justin' )
irc.send ( 'NICK JustinGuese' )
irc.send ( 'JOIN #codingchaos' )
#irc.senc('MSG Peace!')
#irc.send ( 'PRIVMSG #nille :Hello World.\r\n' )

#while True:
#   data = irc.recv ( 4096 )
#   if data.find ( 'PING' ) != -1:
#      irc.send ( 'PONG ' + data.split() [ 1 ] + '\r\n' )
#   print data
